### PR TITLE
feat(kubernetes-fluentd): add Dockerfile

### PR DESCRIPTION
**Pull request by sumo-drosiek**
Signed-off-by: Dominik Rosiek <drosiek@sumologic.com>

### CODE REVIEW

**Comment by sumo-drosiek** [comment]
I have no idea what Im doing

File: kubernetes-fluentd/1.12.2-sumo-0/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,89 @@
+## Use original image to copy files from
+## ref: https://github.com/SumoLogic/sumologic-kubernetes-fluentd/blob/2d2b7693b87f9a0c175740e9674455051c709409/Dockerfile
+FROM public.ecr.aws/sumologic/kubernetes-fluentd:1.12.2-sumo-0 as sumo-image
+
+## Build RedHat compliant image for fluentd
+## ref: https://github.com/fluent/fluentd-docker-image/blob/4e0810850460c7669d7a8ad15d5a3c8ddf28652c/v1.12/debian/Dockerfile
+FROM registry.access.redhat.com/ubi8/ruby-26:1 as fluentd
+
+LABEL maintainer "Fluentd developers <fluentd@googlegroups.com>"
```
Comment (kkujawa-sumo) [inline kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:9]: Maybe this should be set to `collection@sumologic.com`. This is UBI version of image and we will need to maintain this version.

File: kubernetes-fluentd/1.12.2-sumo-0/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+LABEL Description="Fluentd docker image" Vendor="Fluent Organization" Version=1.12.2”
```
Comment (kkujawa-sumo) [inline kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:10]: Maybe vendor should be set to "Sumo Logic" and description corrected to "UBI based Fluentd docker image"

Comment (sumo-drosiek) [inline kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:10]:
```suggestion
```

File: kubernetes-fluentd/1.12.2-sumo-0/Dockerfile (hunk 1 above)
Comment (sumo-drosiek) [reply kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:9]:
```suggestion
```

File: kubernetes-fluentd/1.12.2-sumo-0/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+ && gem install rdoc -v 6.3.1 \
+ && gem cleanup
+COPY --from=sumo-image --chown=fluent:fluent /usr/local/bundle /usr/local/share/gems
+COPY --from=sumo-image /bin/./entrypoint.sh /bin/
+ADD ./snappy.rpm .
+ADD ./snappy-devel.rpm .
+RUN rpm -i snappy.rpm \
+ && rpm -i snappy-devel.rpm \
+ && rm snappy.rpm snappy-devel.rpm
+RUN gem pristine ffi -v 1.15.0 \
```
Comment (pmalek-sumo) [inline kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:83]:
```suggestion
 && rm snappy.rpm snappy-devel.rpm \
 && gem pristine ffi -v 1.15.0 \
```

File: kubernetes-fluentd/1.12.2-sumo-0/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+
+## ref: https://github.com/yashoza19/sumologic-kubernetes-fluentd/blob/c22a932b22ffc591bbc6f2a0ac624c323c1b5006/Dockerfile
+USER root
+RUN yum install -y jq \
+ && gem install rdoc -v 6.3.1 \
+ && gem cleanup
+COPY --from=sumo-image --chown=fluent:fluent /usr/local/bundle /usr/local/share/gems
+COPY --from=sumo-image /bin/./entrypoint.sh /bin/
+ADD ./snappy.rpm .
+ADD ./snappy-devel.rpm .
```
Comment (pmalek-sumo) [inline kubernetes-fluentd/1.12.2-sumo-0/Dockerfile:79]: This can changed to `COPY` since we don't copy remote files

**Review by kkujawa-sumo — APPROVED** [review]

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**